### PR TITLE
Bluetooth: BR: Bt classic auto pts setup

### PR DESCRIPTION
**Pull request by lylezhu2012**
Due to the [PR](https://github.com/zephyrproject-rtos/zephyr/pull/74222) was closed due to the source branch has been deleted. Create a new PR for the changes.

This PR is used to add BR/EDR PTS support to bt_tester.

Test Suite for BR/EDR,
- [x] GAP
- [x] L2CAP
- [x] SMP
- [x] SDP

The changes for auto-pts script is [here](https://github.com/auto-pts/auto-pts/pull/1185)

### CODE REVIEW

File: include/zephyr/bluetooth/hci_types.h [hunk 1]
```diff
@@ -296,6 +296,9 @@ struct bt_hci_cmd_hdr {
 #define BT_HCI_GENERAL_BONDING                  0x04
 #define BT_HCI_GENERAL_BONDING_MITM             0x05
 
+/* Clear Bonding flag */
+#define BT_HCI_SET_NO_BONDING(auth) ((auth) & 0x01)
```
Comment (jhedberg) [inline include/zephyr/bluetooth/hci_types.h:300]: It doesn't seem to me like this needs to be in a public header file. I.e. I'd move it internal to `ssp.c` or just open code it in the single place you're using it (it really doesn't hurt readability IMO).

File: include/zephyr/bluetooth/hci_types.h [hunk 2]
```diff
@@ -299,6 +299,9 @@ struct bt_hci_cmd_hdr {
 /* Clear Bonding flag */
 #define BT_HCI_SET_NO_BONDING(auth) ((auth) & 0x01)
 
+/* Clear MITM flag */
+#define BT_HCI_SET_NO_MITM(auth) ((auth) & (~0x01))
```
Comment (jhedberg) [inline include/zephyr/bluetooth/hci_types.h:303]: Same with this one

File: subsys/bluetooth/host/classic/Kconfig [hunk 3]
```diff
@@ -26,6 +26,12 @@ config BT_MAX_SCO_CONN
 	  Maximum number of simultaneous Bluetooth synchronous connections
 	  supported. The minimum (and default) number is 1.
 
+config BT_L2CAP_CLS
+	bool "Bluetooth L2CAP connectionless data channel [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	help
+	  This option enables Bluetooth L2CAP connectionless data channel
```
Comment (jhedberg) [inline subsys/bluetooth/host/classic/Kconfig:112]: What exactly do you need this feature for? I know it exists in the core spec, but I don't remember any real use cases for it. Let's not add things unless they're needed by some profile for which you have concrete product requirements.

**Comment by jhedberg** [comment]
Please split this up into multiple smaller pull requests. It will be easier to review, and you'll avoid some frustration by being able to see quicker progress (some of the PRs will be possible to merge sooner than others).

File: include/zephyr/bluetooth/hci_types.h (hunk 2 above)
Comment (lylezhu2012) [reply include/zephyr/bluetooth/hci_types.h]: Updated.

**Pull request closed without merging**